flink-ml-lib/src/test/java/org/apache/flink/ml/classification/knn/KnnTest.java

```diff
@@ -0,0 +1,273 @@
+        executeAndCollect(output);
+    }
+
+    @Test
+    public void testFitAndPredict() throws Exception {
+        Knn knn =
+                new Knn()
+                        .setLabelCol(LABEL_COL)
+                        .setFeaturesCol(VEC_COL)
+                        .setK(4)
+                        .setPredictionCol(PRED_COL);
+        KnnModel knnModel = knn.fit(trainData);
+        Table output = knnModel.transform(testData)[0];
+        List<Tuple2<String, String>> result = executeAndCollect(output);
+        verifyClusteringResult(result);
+    }
+
+    @Test
+    public void testSaveLoadAndPredict() throws Exception {
+        String path = Files.createTempDirectory("").toString();
+        Knn knn =
+                new Knn()
+                        .setLabelCol(LABEL_COL)
+                        .setFeaturesCol(VEC_COL)
+                        .setK(4)
+                        .setPredictionCol(PRED_COL);
+        knn.save(path);
+
+        Knn loadKnn = Knn.load(path);
+        KnnModel knnModel = loadKnn.fit(trainData);
+        Table output = knnModel.transform(testData)[0];
+
+        List<Tuple2<String, String>> result = executeAndCollect(output);
+        verifyClusteringResult(result);
+    }
+
+    @Test
+    public void testModelSaveLoadAndPredict() throws Exception {
+        String path = Files.createTempDirectory("").toString();
+        Knn knn =
+                new Knn()
+                        .setLabelCol(LABEL_COL)
+                        .setFeaturesCol(VEC_COL)
+                        .setK(4)
+                        .setPredictionCol(PRED_COL);
+        KnnModel knnModel = knn.fit(trainData);
+        knnModel.save(path);
+        env.execute();
+
+        KnnModel newModel = KnnModel.load(env, path);
+        Table output = newModel.transform(testData)[0];
+        List<Tuple2<String, String>> result = executeAndCollect(output);
+        verifyClusteringResult(result);
+    }
+
+    @Test
+    public void testGetModelData() throws Exception {
+        Knn knn =
+                new Knn()
+                        .setLabelCol(LABEL_COL)
+                        .setFeaturesCol(VEC_COL)
+                        .setK(4)
+                        .setPredictionCol(PRED_COL);
+
+        KnnModel knnModel = knn.fit(trainData);
+        Table modelData = knnModel.getModelData()[0];
+
+        DataStream<Row> output = tEnv.toDataStream(modelData);
+
+        assertEquals(
+                Arrays.asList("VECTORS", "NORM", "LABEL"),
+                modelData.getResolvedSchema().getColumnNames());
+
+        List<Row> modelRows = IteratorUtils.toList(output.executeAndCollect());
+        for (Row modelRow : modelRows) {
+            DenseMatrix vectors = (DenseMatrix) modelRow.getField(0);
+            DenseVector label = (DenseVector) Objects.requireNonNull(modelRow.getField(1));
+
+            assertEquals(2, Objects.requireNonNull(vectors).numRows);
```

Comment:
Could we simplify the code here by removing `Objects.requireNonNull`? It looks like the behavior of this line when `vectors = null` would be the same after we remove it?

In general in Flink codebase we  don't explicitly do `requireNonNull(...)` and it is OK to just throw NPE. Could you check other usages of  `requireNonNull(...)` and follow the existing practice in Flink?

Reply:
OK